File: backend/tests/fixtures/sample.ts
import { useState } from 'react'
import type { Config } from './types'

export interface UserProfile {
  id: string
  name: string
}

export type Status = 'active' | 'inactive'

export enum Color {
  Red = 'red',
  Blue = 'blue',
}

export function processData(input: string): string {
  return input.trim()
}

export const fetchUser = async (id: string) => {
  return { id, name: 'test' }
}

export class DataService {
  process(data: string) {
    return data
  }
}

export default function MainComponent() {
  const [state, setState] = useState(0)
  return state
}

File: backend/tests/fixtures/sample.rs
use std::collections::HashMap;
use std::io::Result;

pub mod utils;

pub struct Config {
    name: String,
    values: HashMap<String, String>,
}

pub enum Status {
    Active,
    Inactive,
    Pending,
}

pub trait Processor {
    fn process(&self, input: &str) -> String;
}

impl Config {
    pub fn new(name: &str) -> Self {
        Config {
            name: name.to_string(),
            values: HashMap::new(),
        }
    }
}

pub fn create_config(name: &str) -> Config {
    Config::new(name)
}

pub async fn fetch_data(url: &str) -> Result<String> {
    Ok(url.to_string())
}
